src/platform/silabs/efr32/rs911x/BLEManagerImpl.cpp

```diff
@@ -0,0 +1,1008 @@
+    int32_t status = 0;
+
+    ChipLogProgress(DeviceLayer, "StartAdvertising start");
+
+    // If already advertising, stop it, before changing values
+    if (mFlags.Has(Flags::kAdvertising))
+    {
+        // sl_bt_advertiser_stop(advertising_set_handle);
+    }
+    else
+    {
+        ChipLogDetail(DeviceLayer, "Start BLE advertissement");
+    }
+
+    err = ConfigureAdvertisingData();
+    SuccessOrExit(err);
+
+    mFlags.Clear(Flags::kRestartAdvertising);
+
+    sl_wfx_mac_address_t macaddr;
+    wfx_get_wifi_mac_addr(SL_WFX_STA_INTERFACE, &macaddr);
+
+    //! Set local name
+    status = rsi_ble_start_advertising();
+    if (status == RSI_SUCCESS)
+    {
+        ChipLogProgress(DeviceLayer, "rsi_ble_start_advertising Success");
+
+        if (mFlags.Has(Flags::kFastAdvertisingEnabled))
+        {
+            StartBleAdvTimeoutTimer(CHIP_DEVICE_CONFIG_BLE_ADVERTISING_INTERVAL_CHANGE_TIME);
+        }
+        mFlags.Set(Flags::kAdvertising);
+    }
+    else
+    {
+        ChipLogProgress(DeviceLayer, "rsi_ble_start_advertising Failed with status: %lx", status);
+    }
+
+exit:
+    ChipLogError(DeviceLayer, "StartAdvertising() End error: %s", ErrorStr(err));
+    return CHIP_NO_ERROR; // err;
+}
+
+// TODO:: Implementation need to be done.
+CHIP_ERROR BLEManagerImpl::StopAdvertising(void)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (mFlags.Has(Flags::kAdvertising))
+    {
+        mFlags.Clear(Flags::kAdvertising).Clear(Flags::kRestartAdvertising);
+        mFlags.Set(Flags::kFastAdvertisingEnabled, true);
+        advertising_set_handle = 0xff;
+        CancelBleAdvTimeoutTimer();
+    }
+
+    // exit:
+    return err;
+}
+
+void BLEManagerImpl::UpdateMtu(rsi_ble_event_mtu_t evt)
+{
+    CHIPoBLEConState * bleConnState = GetConnectionState(event_msg.connectionHandle);
+    if (bleConnState != NULL)
+    {
+        // bleConnState->MTU is a 10-bit field inside a uint16_t.  We're
+        // assigning to it from a uint16_t, and compilers warn about
+        // possibly not fitting.  There's no way to suppress that warning
+        // via explicit cast; we have to disable the warning around the
+        // assignment.
+        //
+        // TODO: https://github.com/project-chip/connectedhomeip/issues/2569
+        // tracks making this safe with a check or explaining why no check
+        // is needed.
+        ChipLogProgress(DeviceLayer, "DriveBLEState UpdateMtu %d", evt.mtu_size);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
+        bleConnState->mtu = evt.mtu_size;
+#pragma GCC diagnostic pop
+        ;
+    }
+}
+
+void BLEManagerImpl::HandleBootEvent(void)
+{
+    mFlags.Set(Flags::kEFRBLEStackInitialized);
+}
+
+void BLEManagerImpl::HandleConnectEvent(void)
+{
+    ChipLogProgress(DeviceLayer, "Connect Event for handle : %d", event_msg.connectionHandle);
+    AddConnection(event_msg.connectionHandle, event_msg.bondingHandle);
+}
+
+// TODO:: Implementation need to be done.
```

Comment:
is this comment still true ? it seems implemented

Reply:
@jmartinez-silabs yeah it is implemented so removed the comment.